portfolio/src/main/webapp/script.js

```diff
@@ -12,26 +12,32 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var factIndex = 0;
```

Comment:
The JavaScript style guide says to never use `var` (it has very strange behavior because it's function-scoped), so please change this to `let`. Same below for favoriteIndex.

Reply:
Replying so that I can request changes in the review.